src/frog-base.tsx

```diff
@@ -168,9 +169,22 @@ export type FrogConstructorParameters<
   verify?: boolean | 'silent' | undefined
 }
 
-export type RouteOptions = Pick<FrogConstructorParameters, 'verify'> & {
-  fonts?: ImageOptions['fonts'] | (() => Promise<ImageOptions['fonts']>)
-}
+export type RouteOptions<M extends string = string> = Pick<
+  FrogConstructorParameters,
+  'verify'
+> &
+  (M extends 'frame'
+    ? {
+        fonts?: ImageOptions['fonts'] | (() => Promise<ImageOptions['fonts']>)
+      }
+    : M extends 'cast-action'
```

Comment:
```suggestion
    : M extends 'castAction'
```

Would be good if it mapped to the casing of the method.

Reply:
done!